EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+The Ethereum Virtual Machine executes all smart contract code on ethereum. In order to easier debug smart contracts and state tests, a de-facto standard was introduced to log every execution step of the EVM. This standard was implemented by go-ethereum, parity and nethermind and Besu. Since the de-facto standard was not well defined, the implementations differed slightly making it hard to develop adequate tooling.
+This EIP has multiple objectives:
+- Move the specification to a more visible place to encourage new clients to implement it
+- Strictly define corner cases that were not addressed in the previous version
+- Allow for updates to the specification in case new fields are introduced during execution
+- Provide sample output
+
+Implementing this EIP in all major clients allows us to create meaningful differential fuzzers that fuzz EVM implementations for the mainnet and all upcoming hardforks.
+It also helps finding differences in execution quickly in the case of a chain split.
+
+Due to the slight differences in output the following fields currently need to be dropped when comparing tracing outputs: $GasCost$, $MemorySize$, $RefundCounter$, $ReturnStack$, and $ReturnData$. This reduces the usefulness of tracing significantly. Creating a common standard would allow to add these fields thus increasing the surface that can be tested by a fuzzer.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Clients should be able to execute simple transactions as well as code and return traces. In the following, we will call this client CUT (client under test) and use go-ethereums `evm` binary for code examples. 
+
+### Inputs
+
+The CUT MUST be able to execute a statetest as described [here](https://www.ethdocs.org/en/latest/contracts-and-transactions/ethereum-tests/state_tests/index.html).
+Example :
+``` 
+evm --json --nomemory statetest foobar.json 
+```
+The CUT MUST spit out jsonl as defined below, and it MUST also report the final `stateRoot` hash after executing the test. The CUT SHOULD report whether the tests succeeded or failed.
+
+If the statetest contains a Fork identifier (e.g. `Berlin` or `Istanbul`) the CUT MUST run the statetest under the chain rules of this fork.
+
+The CUT SHOULD executed statetests if the filenames are written to STDIN. This allows to execute multiple tests on the same instance without having to set up and tear down the client.
+
+### Output
```

Comment:
Rather than output this would be the trace objects.  Some expository about how this would be an array of objects if used in a wrapping json object like a JSON-RPC call should be added.